Build issue: Build with clang 16 fails with undefined template 'std::basic_ostringstream<char>' error

### Build error message(s)

When building GridKit with clang 16 I get following build error:
```console
$ make

...

Building CXX object GridKit/Model/PhasorDynamics/Branch/CMakeFiles/phasor_dynamics_branch.dir/BranchEnzyme.cpp.o
In file included from /Users/55y/src/gridkit/GridKit/GridKit/Model/PhasorDynamics/Branch/BranchEnzyme.cpp:9:
In file included from /Users/55y/src/gridkit/GridKit/GridKit/Model/PhasorDynamics/Branch/BranchImpl.hpp:16:
/Users/55y/src/gridkit/GridKit/GridKit/Model/VariableMonitorImpl.hpp:190:28: error: implicit instantiation of undefined template 'std::basic_ostringstream<char>'
        std::ostringstream v_os;
                           ^
/opt/local/libexec/llvm-16/bin/../include/c++/v1/iosfwd:131:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_ostringstream;
```


### GridKit™ version

`develop` at 8f542530219c8b2ad515dfc27af80a6766969a6d

### System and environment details

- MacOS 15.7.3
- clang version 16.0.6 from MacPorts
- arm cpu

### Additional information

The issue does not show up when using GCC.

--- a/GridKit/Model/VariableMonitor.hpp
+++ b/GridKit/Model/VariableMonitor.hpp
@@ -9,6 +9,7 @@
 #include <limits>
 #include <memory>
 #include <set>
+#include <sstream>
 #include <string>
 #include <type_traits>
 #include <variant>

